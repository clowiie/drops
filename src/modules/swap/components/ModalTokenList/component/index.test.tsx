import { render } from '@testing-library/react'

import mockComponent from 'test/utils/mockComponent'

import * as ModalTokenListType from '.'

describe('ModalTokenList', () => {
  const ModalSpy = mockComponent('common/components/Modal')
  const SwapTokenListSpy = mockComponent(
    'modules/swap/components/SwapTokenList',
  )

  const onCloseSpy = jest.fn()
  const getTokenSpy = jest.fn()

  const { default: ModalTokenList } = require('.') as typeof ModalTokenListType

  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('snapshots', () => {
    it('should match snapshots', () => {
      const { container, getByText } = render(
        <ModalTokenList isOpen onClose={onCloseSpy} getToken={getTokenSpy} />,
      )

      expect(container.firstChild).toMatchSnapshot()
      expect(ModalSpy).toBeCalledTimes(1)
      expect(SwapTokenListSpy).toBeCalledTimes(1)
      expect(SwapTokenListSpy).toBeCalledWith({
        getToken: getTokenSpy,
      })
      expect(getByText('Select Token')).toBeDefined()
    })
  })
})
