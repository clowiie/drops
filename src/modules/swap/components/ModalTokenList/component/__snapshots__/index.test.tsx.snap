// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalTokenList snapshots should match snapshots 1`] = `
<div>
  <div
    class="py-4"
  >
    <div
      class="px-4"
    >
      <span
        class="text-lg font-semibold"
      >
        Select Token
      </span>
    </div>
    <div
      class="pt-4"
    >
      <div />
    </div>
  </div>
</div>
`;
